python/ray/import_thread.py

```diff
@@ -136,9 +136,13 @@ def _process_key(self, key):
                         ray_constants.DUPLICATE_REMOTE_FUNCTION_THRESHOLD)
 
         if key.startswith(b"RemoteFunction"):
-            with profiling.profile("register_remote_function"):
-                (self.worker.function_actor_manager.
-                 fetch_and_register_remote_function(key))
+            # TODO (Alex): There's a race condition here if the worker is
+            # shutdown before the function finished registering (because core
+            # worker's global worker is unset before shutdown and is needed
+            # for profiling).
+            # with profiling.profile("register_remote_function"):
+            (self.worker.function_actor_manager.
+             fetch_and_register_remote_function(key))
```

Comment:
Can you add a unit test?

Reply:
```
import ray

ray.init(num_cpus=1)


jobs = []
for _ in range(8):

    @ray.remote(max_calls=1)
    def foo():
        return

    jobs.append(foo.remote())
ray.get(jobs)
```

This snippet seems to do it.

Reply:
We discussed this, but just writing it here. Not adding a unit test because it's not worth it. This bug is actually quite minor because only causes the worker to die after shutdown is called (therefore it doesn't meaningfully effect the program's behavior). Not worth parsing stderr for the right error message.

Reply:
@wuisawesome is there another way to fix this aside from just disabling the profiling? We basically just need to make sure import thread stops before the core worker is destructed, right? Not a huge deal but would be better to keep this in the timeline.

Reply:
Stopping the thread from C++ seems pretty hard, I don't think it's really worth it for now.